lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -168,7 +168,18 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('user_id', request.user.id)
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
 
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
         course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        is_enrolled = enrollment and enrollment.is_active
+        is_staff = bool(has_access(request.user, 'staff', course_key))
+        show_enrolled = is_enrolled or is_staff
+
+        # Move retrieving the course outline blocks before we set up masquerade
```

Comment:
So I understand moving this call, but I noticed you left all of the other functionality that also uses the request.user. Should all of those also be pulled up?

I guess the larger question is what exactly should be viewed as the masquerading user and what shouldn't be? We are deciding staff users should be able to see the full outline, but as it stands in this PR right now, they will be limited in what they see for dates in the sidebar and other functionality that uses request.user.

Reply:
i was trying to limit the scope of the change to only the thing requested in the bug but i took a look and moving the call to the end should be ok, so for now i've done that and posted in slack asking kaitlin for clarification on what we want

Reply:
Wouldn't this new location break the flow for masquerading as a specific learner though? Because in this case, you'll only see the course as your staff user and never the masqueraded user. e.g. Completeness information.

Sorry, I know this whole masquerade business is quite complicated and messy.

Reply:
[passing comment; I'm not requesting any changes]

> I guess the larger question is what exactly should be viewed as the masquerading user and what shouldn't be?

I struggled with this a bit when working on the courseware-side version of this bug. I'm not sure we have a good answer right now, other than "whatever generates the fewest partner support tickets" :grimacing: Whoever becomes owners of Masquerade after the T&L split would behoove themselves to figure out a high-level principle for what masquerading should change.

FWIW, my working principle has been:
* top-level course access **shouldn't be** affected by masquerade (so that Staff viewing as Learners can still see courses in which they're Staff, but not enrolled in).
* content-level course access and behavior **should be** affected by masquerade (so that Staff viewing as Learners can experience the course like a Learner would). 

Unfortunately, those two bullets are not as cleanly distinguished as we'd want them to be, so it's not always possible or reasonable to follow that advice.

Reply:
I asked Kaitlin in slack https://edx-internal.slack.com/archives/G019J2RMNEA/p1617281028008600 and she said to show the non-user-specific content, so i've updated the PR to do this.  
i changed the code to be a less DRY but more readable i think